src/Vizkit3dWorld.cpp

```diff
@@ -330,4 +345,34 @@ void Vizkit3dWorld::applyCameraParams() {
     widget->getView(0)->getCamera()->setProjectionMatrixAsPerspective(osg::RadiansToDegrees(fovy), aspectRatio, zNear, zFar);
 }
 
+void Vizkit3dWorld::removeLinksChild(sdf::ElementPtr sdf_model, std::string child_element){
+
+    sdf::ElementPtr linkElem = sdf_model->GetElement("link");
+
+    while(linkElem){
+        if (linkElem->HasElement(child_element)){
+            sdf::ElementPtr undesired_elem = linkElem->GetElement(child_element);
+
+            //stores all the undesired elements before delete them
+            std::vector<sdf::ElementPtr> erase_vector;
+            while (undesired_elem)
+            {
+                erase_vector.push_back(undesired_elem);
+
+                undesired_elem = undesired_elem->GetNextElement(child_element);
+
```

Comment:
And definitely do NOT add empty spaces before a block end / after a block beginning.